security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,345 @@
+
+Still within the busybox "access" pod, issue the following command to test access to google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It should return the HTML of the google.com home page.
+
+### 2. Deny all ingress traffic
+
+Enable ingress isolation on the namespace by deploying a [default deny all ingress traffic policy](https://docs.projectcalico.org/security/kubernetes-default-deny).
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: default-deny-ingress
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  ingress:
+  - action: Deny
+EOF
+```
+
+#### Verify access - denied all ingress and allowed all egress
+
+Because all pods in the namespace are now selected, any ingress traffic which is not explicitly allowed by a policy will be denied.
+
+We can see that this is the case by switching over to our "access" pod in the namespace and attempting to access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It should return:
+
+```
+wget: download timed out
+```
+
+{: .no-select-button}
+
+Next, try to access google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It should return:
+
+```
+<!doctype html><html itemscope="" item....
+```
+
+{: .no-select-button}
+
+We can see that the ingress access to the nginx service is denied while egress access to outbound internet is still allowed.
+
+### 3. Allow ingress traffic to Nginx
+
+Run the following to create a `NetworkPolicy` which allows traffic to nginx pods from any pods in the `advanced-policy-demo` namespace.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: access-nginx
+  namespace: advanced-policy-demo
+spec:
+  selector: run == 'nginx'
+  ingress:
+  - action: Allow
+    source:
+      selector: all()
+EOF
+```
+
+#### Verify access - allowed nginx ingress
+
+Now ingress traffic to nginx will be allowed. We can see that this is the case by switching over to our "access" pod in the namespace and attempting to access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It should return:
+
+```
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>...
+```
+
+{: .no-select-button}
+
+After creating the policy, we can now access the nginx Service.
+
+### 4. Deny all egress traffic
+
+Enable egress isolation on the namespace by deploying a [default deny all egress traffic policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/#4-deny-all-egress-traffic).
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: default-deny-egress
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  egress:
+  - action: Deny
+EOF
+```
+
+#### Verify access - denied all egress
+
+Now any ingress or egress traffic which is not explicitly allowed by a policy will be denied.
```

Comment:
Now any ingress or egress traffic **that** is not explicitly allowed by a policy will be denied.

Reply:
Done! Thank you